server/graphql/types/src/types/purchase_order.rs

```diff
@@ -0,0 +1,222 @@
+use self::dataloader::DataLoader;
+use async_graphql::*;
+use chrono::{DateTime, NaiveDate, NaiveDateTime, Utc};
+use graphql_core::loader::{
+    NameByIdLoader, NameByIdLoaderInput, PurchaseOrderLinesByPurchaseOrderIdLoader,
+    StoreByIdLoader, UserLoader,
+};
+use graphql_core::ContextExt;
+use repository::{PurchaseOrderRow, PurchaseOrderStatus};
+use service::ListResult;
+
+use crate::types::{NameNode, PurchaseOrderLineConnector, StoreNode, UserNode};
+
+#[derive(PartialEq, Debug)]
+pub struct PurchaseOrderNode {
+    pub purchase_order: PurchaseOrderRow,
+}
+#[derive(SimpleObject)]
+pub struct PurchaseOrderConnector {
+    pub total_count: u32,
+    pub nodes: Vec<PurchaseOrderNode>,
+}
+
+#[Object]
+impl PurchaseOrderNode {
+    pub async fn id(&self) -> &str {
+        &self.row().id
+    }
+    pub async fn store(&self, ctx: &Context<'_>) -> Result<Option<StoreNode>> {
+        let loader = ctx.get_loader::<DataLoader<StoreByIdLoader>>();
+        Ok(loader
+            .load_one(self.row().store_id.clone())
+            .await?
+            .map(StoreNode::from_domain))
+    }
+    pub async fn user(&self, ctx: &Context<'_>) -> Result<Option<UserNode>> {
+        let loader = ctx.get_loader::<DataLoader<UserLoader>>();
+
+        let result = loader
+            .load_one(self.row().user_id.clone())
+            .await?
+            .map(UserNode::from_domain);
+
+        Ok(result)
+    }
+    pub async fn supplier_name_link_id(&self) -> &Option<String> {
+        &self.row().supplier_name_link_id
+    }
+    pub async fn supplier(&self, ctx: &Context<'_>, store_id: String) -> Result<Option<NameNode>> {
+        let loader = ctx.get_loader::<DataLoader<NameByIdLoader>>();
+
+        let result = match &self.row().supplier_name_link_id {
+            Some(id) => loader
+                .load_one(NameByIdLoaderInput::new(&store_id, id))
+                .await?
+                .map(NameNode::from_domain),
+            None => None,
+        };
+
+        Ok(result)
+    }
+    pub async fn purchase_order_number(&self) -> &i32 {
+        &self.row().purchase_order_number
+    }
+    pub async fn created_datetime(&self) -> DateTime<Utc> {
+        DateTime::<Utc>::from_naive_utc_and_offset(self.row().created_datetime, Utc)
+    }
+    pub async fn delivered_datetime(&self) -> &Option<NaiveDateTime> {
+        &self.row().delivered_datetime
+    }
```

Comment:
Would this be a `DateTime<Utc>` like the `created_datetime`? It's the same type in the database?

Reply:
Yes good point thanks Carl